tag.go

```diff
@@ -5,16 +5,25 @@ import "strings"
 type tagValue struct {
```

Comment:
See above: should the type and fields be exported? Also we can now remove `optional`.

I also think, but up to discussion, that the flag type could just be a string.

We could also had a function (and keep `flags` private):

```go
func (t TagValue) HasFlag(f string) bool {
 _, ok := t.flags[f]
return ok
}
```

Reply:
I agree about the flags being strings, I was 50-50 on the typed string being overkill, the `HasFlag` approach is simpler/better.